src/components/StatePicker.js

```diff
@@ -14,31 +15,64 @@ const propTypes = {
     /** The label for the field */
     label: PropTypes.string,
 
-    /** A callback method that is called when the value changes and it received the selected value as an argument */
-    onChange: PropTypes.func.isRequired,
+    /** A callback method that is called when the value changes and it receives the selected value as an argument. */
+    onInputChange: PropTypes.func.isRequired,
 
     /** The value that needs to be selected */
     value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
 
+    /** Indicates that the input is being used with the Form component */
+    isFormInput: PropTypes.bool,
+
+    /**
+    * The ID used to uniquely identify the input
+    *
+    * @param {Object} props - props passed to the input
+    * @returns {Object} - returns an Error object if isFormInput is supplied but inputID is falsey or not a string
+    */
+    inputID: props => FormUtils.validateInputIDProps(props),
+
+    /** Saves a draft of the input value when used in a form */
+    shouldSaveDraft: PropTypes.bool,
+
+    /** Callback that is called when the text input is blurred */
+    onBlur: PropTypes.func,
+
+    /** Error text to display */
+    errorText: PropTypes.string,
+
+    /** The default value of the state picker */
+    defaultValue: PropTypes.string,
+
     ...withLocalizePropTypes,
 };
 
 const defaultProps = {
     label: '',
-    value: '',
+    value: undefined,
+    errorText: '',
+    shouldSaveDraft: false,
+    isFormInput: false,
+    inputID: undefined,
+    onBlur: () => {},
 };
 
-const StatePicker = props => (
+const StatePicker = forwardRef((props, ref) => (
     <Picker
+        ref={ref}
+        inputID={props.inputID}
         placeholder={{value: '', label: '-'}}
         items={STATES}
-        onInputChange={props.onChange}
+        onInputChange={props.onInputChange}
         value={props.value}
+        defaultValue={props.defaultValue}
```

Comment:
I think we should pass either `value` or `defaultValue` but not both.

Reply:
now added conditional value here, if the value is available then the value will be used otherwise defaultValue